File: bindings/nodejs/examples/client/get-validators.ts
// Copyright 2024 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Client, initLogger } from '@iota/sdk';
require('dotenv').config({ path: '.env' });

// Run with command:
// yarn run-example ./client/get-validators.ts [PAGE_SIZE] [CURSOR]

// This example returns the validators known by the node by querying the corresponding endpoint.
// You can provide a custom PAGE_SIZE and additionally a CURSOR from a previous request.
async function run() {
    initLogger();
    for (const envVar of ['NODE_URL']) {
        if (!(envVar in process.env)) {
            throw new Error(`.env ${envVar} is undefined, see .env.example`);
        }
    }

    const client = await Client.create({
        // Insert your node URL in the .env.
        nodes: [process.env.NODE_URL as string],
    });

    let pageSize = 1;
    let cursor = '';
    if (process.argv.length > 1) {
        pageSize = parseInt(process.argv[2]);
        if (process.argv.length > 2) {
            cursor = process.argv[3];
        }
    }

    try {
        const validators = await client.getValidators(pageSize, cursor);
        console.log(validators);
    } catch (error) {
        console.error('Error: ', error);
    }
}

void run().then(() => process.exit());

File: bindings/python/examples/client/get_validators.py
import dataclasses
import json
import os
import sys

from dotenv import load_dotenv
from iota_sdk import Client

load_dotenv()

node_url = os.environ.get('NODE_URL', 'https://api.testnet.shimmer.network')
page_size = 1
cursor = ""

if len(sys.argv) > 1:
    page_size = int(sys.argv[1])
    if len(sys.argv) > 2:
        cursor = sys.argv[2]

# Create a Client instance
client = Client(nodes=[node_url])

validators = client.get_validators(page_size, cursor)
print(f'{json.dumps(dataclasses.asdict(validators), indent=4)}')
